dpcpp/CMakeLists.txt

```diff
@@ -51,11 +51,15 @@ target_sources(ginkgo_dpcpp
 
 ginkgo_compile_features(ginkgo_dpcpp)
 
-set(GINKGO_DPCPP_FLAGS ${GINKGO_COMPILER_FLAGS} -fsycl -fsycl-device-lib=libc)
+set(GINKGO_DPCPP_FLAGS ${GINKGO_COMPILER_FLAGS} -fsycl)
 set(GINKGO_DPCPP_FLAGS ${GINKGO_DPCPP_FLAGS} PARENT_SCOPE)
 target_compile_options(ginkgo_dpcpp PRIVATE "${GINKGO_DPCPP_FLAGS}")
 target_compile_features(ginkgo_dpcpp PRIVATE cxx_std_17)
+target_link_options(ginkgo_dpcpp PRIVATE -fsycl-device-lib=all)
```

Comment:
Should we provide option for DPCPP_JIT_OPTION or DPCPP_LINK_OPTION?
With specific JIT_OPTION can allow us to compile full ginkgo with double support but does not face the double emulation issue on float-only device

Reply:
I do not understand what this is, what option do you mean from the compiler?

Reply:
`-fsycl-device-code-split=per_kernel` allow us to compile ginkgo dpcpp with double kernel. It only gives error when the program really use the kernel with double.